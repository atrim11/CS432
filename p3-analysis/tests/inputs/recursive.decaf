def int dangerous_recursion(int n) {
    if (n > 0) {
        return dangerous_recursion(n + 1);
    }
    return 0;
}

def int main() {
    return dangerous_recursion(5);
}
