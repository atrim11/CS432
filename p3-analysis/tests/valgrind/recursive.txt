==404907== Memcheck, a memory error detector
==404907== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==404907== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==404907== Command: ../decaf inputs/recursive.decaf
==404907== 
Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 dangerous_recursion : (int) -> int
 main : () -> int

  FuncDecl name="dangerous_recursion" return_type=int parameters={n:int} [line 1]
  SYM TABLE:
   n : int

    Block [line 1]
    SYM TABLE:

        Block [line 2]
        SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 8]
  SYM TABLE:

    Block [line 8]
    SYM TABLE:

==404907== 
==404907== HEAP SUMMARY:
==404907==     in use at exit: 0 bytes in 0 blocks
==404907==   total heap usage: 1,968 allocs, 1,968 frees, 235,448 bytes allocated
==404907== 
==404907== All heap blocks were freed -- no leaks are possible
==404907== 
==404907== For lists of detected and suppressed errors, rerun with: -s
==404907== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
