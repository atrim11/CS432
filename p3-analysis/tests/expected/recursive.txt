Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 dangerous_recursion : (int) -> int
 main : () -> int

  FuncDecl name="dangerous_recursion" return_type=int parameters={n:int} [line 1]
  SYM TABLE:
   n : int

    Block [line 1]
    SYM TABLE:

        Block [line 2]
        SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 8]
  SYM TABLE:

    Block [line 8]
    SYM TABLE:

